ext/socket/lib/socket.rb

```diff
@@ -623,45 +684,131 @@ def accept_nonblock(exception: true)
   def self.tcp(host, port, local_host = nil, local_port = nil, connect_timeout: nil, resolv_timeout: nil) # :yield: socket
```

Comment:
Better there be an option added to force protocol family to 4 or 6?

Reply:
I feel that's good, but it will be an API change.
I'd like to focus on improving it without API chnages.

Reply:
It would be a backward compatible change so it shouldn't break anything. Especially if the change is using named parameter as adding more positional parameters will be ugly and unnecessary.

On the other hand, if such parameter is missing, there might be users that see incompatible behavior and can't modify it.